### ISSUE TITLE
adapt ng5: http: deprecate @angular/http in favor of @angular/common/http (#18906) (72c7b6e)

### ISSUE THREAD

**Issue by zijianhuang** [zijianhuang/webapiclientgen#21]
Breaking changes:

Class names changed. Not a big deal.
```js
//import { Http, Headers, Response } from '@angular/http';
import { HttpClient, HttpHeaders, HttpResponse } from '@angular/common/http';
```

In angular/http:
```js
        getDecimalSquare(d: number): Observable<number>{
            return this.http.get(this.baseUri + 'api/SuperDemo/decimal?d='+d).map(response=> response.json());
        }
```

In angular/common/http
```js
        getIntSquare(d: number): Observable<number> {
            return this.http.get<number>(this.baseUri + 'api/SuperDemo/int?d=' + d);
        }
```

The new lib makes good use of generic and hide more technical details behind. Higher abstraction. 

So the generated codes may look shorter.

**Comment by zijianhuang** [comment]
basically done and tested.

Ready to release after testing in a commercial project.

**Comment by zijianhuang** [comment]
released as 2.3
